app/code/Magento/Customer/Block/Form/Register.php

```diff
@@ -220,4 +244,35 @@ public function getRequiredCharacterClassesNumber()
     {
         return $this->_scopeConfig->getValue(AccountManagement::XML_PATH_REQUIRED_CHARACTER_CLASSES_NUMBER);
     }
+
+    /**
+     * Determine if Address Fields have to be shown
+     *
+     * @return bool
+     */
+    public function getShowAddressFields()
+    {
+        return $this->_scopeConfig->getValue(self::XML_PATH_REGISTER_SHOW_ADDRESS_FIELDS);
+    }
+
+    /**
+     * Return form attribute IDs by form code
+     *
+     * @param string $formCode
+     * @return array
+     */
+    public function isAddressFieldUsedInForm($attributeCode)
+    {
+        $formCode = 'customer_register_address';
+        $entityType = AddressMetadataInterface::ENTITY_TYPE_ADDRESS;
+        $attributeId = $this->_eavAttribute->getIdByCode( $entityType, $attributeCode);
+        $bind = ['attribute_id' => $attributeId, 'form_code' => $formCode];
+        $select = $this->_resource->getConnection()->select()->from(
```

Comment:
It is not a recommended approach having SQL queries in the Blocks. Ideally they should be placed in resource models.